roseau/load_flow/models/buses.py

```diff
@@ -51,58 +50,64 @@ def __init__(
         super().__init__(id, **kwargs)
         self._check_phases(id, phases=phases)
         self.phases = phases
-        n = len(phases)
         if potentials is None:
-            potentials = np.zeros(n, dtype=complex)
             self.initialized = False
+            potentials = [0] * len(phases)
         else:
-            if len(potentials) != n:
-                msg = f"Incorrect number of potentials: {len(potentials)} instead of {n}"
-                logger.error(msg)
-                raise RoseauLoadFlowException(msg=msg, code=RoseauLoadFlowExceptionCode.BAD_POTENTIALS_SIZE)
-            if isinstance(potentials, Quantity):
-                potentials = potentials.m_as("V")
             self.initialized = True
-        self.initial_potentials = np.asarray(potentials)
+        self.potentials = potentials
         self.geometry = geometry
-        self._potentials = None
+
+        self._res_potentials: Optional[list[complex]] = None
 
     def __repr__(self) -> str:
-        s = f"{type(self).__name__}(id={self.id!r}, phases={self.phases!r}"
-        if self._potentials is not None:
-            s += f", potentials={self.potentials!r}"
-        if self.geometry is not None:
-            s += f", geometry={self.geometry}"
-        s += ")"
-        return s
+        return f"{type(self).__name__}(id={self.id!r}, phases={self.phases!r})"
+
+    def _validate_potentials(self, potentials: Sequence[complex]) -> list[complex]:
+        if len(potentials) != len(self.phases):
+            msg = f"Incorrect number of potentials: {len(potentials)} instead of {len(self.phases)}"
+            logger.error(msg)
+            raise RoseauLoadFlowException(msg=msg, code=RoseauLoadFlowExceptionCode.BAD_POTENTIALS_SIZE)
+        return list(potentials)
```

Comment:
It wasn't consistent. The type was dependent on the user providing a list or array in *most* places, so I made them all lists for consistency. I can make them all arrays instead but I chose list because the usage was always through the list interface (in loops or comprehensions). I have no strong preference for list vs. array but I think we should a have a unified interface.

Reply:
In this case, the user input must accept `Sequence[complex]`.  The output and internal fields should be numpy array